core/services/chainlink/config.go

```diff
@@ -52,6 +52,26 @@ func (c *Config) TOMLString() (string, error) {
 	return string(b), nil
 }
 
+// warnings aggregates warnings from valueWarnings and deprecationWarnings
+func (c *Config) warnings() (err error) {
+	_, warning := utils.MultiErrorList(c.deprecationWarnings())
+	err = c.valueWarnings()
+	err = multierr.Append(warning, err)
```

Comment:
should we append before calling MultiErrorList? Or call it separately for valueWarning before appending?

Reply:
This is the user experience when running validate with a value warning and a deprecation warning, lgtm:
```
# Configuration warning:
2 errors:
	- P2P.V1: is deprecated and will be removed in a future version
	- Tracing.TLSCertPath: invalid value (something): must be empty when Tracing.Mode is 'insecure'
```

Reply:
Is there a test that demonstrates this? Shouldn't this show in `Test_generalConfig_LogConfiguration`?

Reply:
No, since deprecation warnings are in the full config fixture, which isn't tested in `Test_generalConfig_LogConfiguration`. I wrote a unit test to cover `warnings()`.

Reply:
I've also added a testscripts test - there's one for deprecation warnings already so just added on to that one.

Reply:
Your intuition was right:

```
                2 errors:
                	- 2 errors:
                	- P2P.V1: is deprecated and will be removed in a future version
                	- P2P.V1: 10 errors:
                		- AnnounceIP: is deprecated and will be removed in a future version
                		- AnnouncePort: is deprecated and will be removed in a future version
                		- BootstrapCheckInterval: is deprecated and will be removed in a future version
                		- DefaultBootstrapPeers: is deprecated and will be removed in a future version
                		- DHTAnnouncementCounterUserPrefix: is deprecated and will be removed in a future version
                		- DHTLookupInterval: is deprecated and will be removed in a future version
                		- ListenIP: is deprecated and will be removed in a future version
                		- ListenPort: is deprecated and will be removed in a future version
                		- NewStreamTimeout: is deprecated and will be removed in a future version
                		- PeerstoreWriteInterval: is deprecated and will be removed in a future version
                	- Tracing.TLSCertPath: invalid value (something): must be empty when Tracing.Mode is 'unencrypted'
 ```
 
 I'll fix it - thanks for the comment